integration-tests/runtime-tests/src/generators/crowdloanGenerators/crowdloanRewardGenerator.ts

```diff
@@ -0,0 +1,105 @@
+import { ApiPromise, Keyring, WsProvider } from '@polkadot/api';
+import { KeyringPair } from '@polkadot/keyring/types';
+import minimist from 'minimist';
+import * as R from 'ramda';
+import { args } from '../../utils/args';
+
+/**
+ * Contains all generation methods as static async functions
+ * 
+ * WARN: await Delay can cause problems!
+**/
+export class crowdloanRewardGenerator {
+  /**
+   * 
+   * @param { ApiPromise } api Connected API Promise.
+   * @param { KeyringPair } sudoKey with sudo access.
+   * @param { KeyringPair } walletAlice 
+   */
+  public static async testCrowdloanRewards(api: ApiPromise, sudoKey: KeyringPair, walletAlice: KeyringPair) {
+    /*
+      Populate
+      Initialize
+      Associate (automatically execute first claim)
+    */
+
+    const toHexString = bytes =>
+      Array.prototype.map.call(bytes, x => ('0' + (x & 0xFF).toString(16)).slice(-2)).join('');
+
+    const delay = () => new Promise(res => setTimeout(res, 24_000));
```

Comment:
probably better to add that warn here

Reply:
Makes sense. Though this file will be reworked to wait for results instead of using delay, similar to the "txCrowdloanRewardsTests.ts" file.